### PR TITLE
Fix template codegen with static parameters

### LINKED ISSUE #7477: Wrapping CPP with static[int] generates wrong code
```nim
type
  SmallString* {.importcpp: "llvm::SmallString", header: "llvm/ADT/SmallString.h", noDecl.} [N: static[int]] = object

var x: SmallString[128]
```

Compiled with: ``nim cpp --passC:"-I/usr/local/opt/llvm/include" --passC:"-std=c++11" file``

Error:

```
/private/tmp/aporia/nimcache/a4.cpp:24:27: error: template argument for non-type template parameter must be an expression
typedef llvm::SmallString<NI>  TY_7XJAavvgqEKfZ9bvblIL8mw;
                          ^~
/usr/local/opt/llvm/include/llvm/ADT/SmallString.h:25:19: note: template parameter is declared here
template<unsigned InternalLen>
                  ^
1 error generated.
```

### PR DESCRIPTION
**Pull request by LemonBoy**
Fixes #7477

### CODE REVIEW

File: compiler/ccgtypes.nim [hunk 1]
```diff
@@ -734,6 +734,16 @@ proc getTypeDescAux(m: BModule, origTyp: PType, check: var IntSet): Rope =
       let cppName = getTypeName(m, t, sig)
       var i = 0
       var chunkStart = 0
+
+      template addResultType(ty: untyped) =
```
Comment (zah) [inline compiler/ccgtypes.nim]: This deserves to be a regular proc IMO.

File: tests/cpp/t7477.nim [hunk 2]
```diff
@@ -0,0 +1,11 @@
+{.emit: """/*TYPESECTION*/
```
Comment (zah) [inline tests/cpp/t7477.nim]: Move this to `tstaticimportcpp`.

You can use a type declaration such as `T data[N];` to make the code compatible with the older C++ standards.

File: compiler/ccgtypes.nim (hunk 1 above)
Comment (LemonBoy) [reply compiler/ccgtypes.nim]: The code isn't used at all outside of this proc, I believe a template hits the sweet spot between a proc and copy&paste.

File: tests/cpp/t7477.nim (hunk 2 above)
Comment (LemonBoy) [reply tests/cpp/t7477.nim]: Done

File: compiler/ccgtypes.nim [hunk 3]
```diff
@@ -734,6 +734,16 @@ proc getTypeDescAux(m: BModule, origTyp: PType, check: var IntSet): Rope =
...
+        if ty == nil or ty.kind == tyVoid:
+          result.add(~"void")
+        elif ty.kind == tyStatic:
+          internalAssert m.config, ty.n != nil
+          result.add ty.n.renderTree
```
Comment (Araq) [inline compiler/ccgtypes.nim]: Er ... what?! so ``renderTree``  produces valid C code now? Dubious.

Comment (Araq) [reply compiler/ccgtypes.nim]: I see this already was in the original code. Well. @zah you better explain this.

**Pull request merged**